csharp.test/TestMaps.cs

```diff
@@ -49,6 +49,15 @@ public static void CanRoundtripRequiredMaps()
             DoRoundtripTest(false, keys, values);
         }
 
+        [Test]
+        public static void CanRoundtripLogicalMapType()
+        {
+            var keys = new[] {new[] {"k1", "k2"}, new[] {"k3", "k4"}, Array.Empty<string>()};
+            var values = new[] {new[] {"v1", "v2"}, new[] {"v3", "v4"}, Array.Empty<string>()};
+
+            DoRoundtripTest(false, keys, values, CreateMapSchema(false));
```

Comment:
```suggestion
            DoRoundtripTest(false, keys, values, CreateMapSchema(true));
```

Reply:
Oh, that was a little accident when I decided to refactor the existing `CreateMapSchema` instead of writing a new one. It's supposed to be `CreateMapSchema(false, true)`, where the second parameter is the new `nested` option.

Reply:
Oh right, sorry I misread that. Looks good now